Tests on test_services_types.py fail randomly on Jenkins

I noticed that the tests at [test_services_types.py](https://github.com/plone/plone.restapi/blob/master/src/plone/restapi/tests/test_services_types.py) are failing with considerable frequency in Jenkins. Errors like:

Error message:
```
500 != 204
```

Traceback
```
500 != 204

  File "/srv/python2.7/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/jenkins/workspace/pull-request-5.2/src/plone.restapi/src/plone/restapi/tests/test_services_types.py", line 216, in test_types_document_patch_fieldsets
    self.assertEqual(response.status_code, 204)
  File "/srv/python2.7/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/srv/python2.7/lib/python2.7/unittest/case.py", line 506, in _baseAssertEqual
    raise self.failureException(msg)
```

For example:

https://jenkins.plone.org/job/pull-request-5.2/1712/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_get_field/

https://jenkins.plone.org/job/pull-request-5.2/1714/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_get_fieldset/

https://jenkins.plone.org/job/pull-request-5.2/1729/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_patch_fieldsets/

https://jenkins.plone.org/job/pull-request-5.2/1730/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_patch_create_missing/

https://jenkins.plone.org/job/pull-request-5.2/1736/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_get_field/

https://jenkins.plone.org/job/pull-request-5.2/1737/testReport/junit/plone.restapi.tests.test_services_types/TestServicesTypes/test_types_document_patch_create_missing/

These errors are likely due to high load on the Jenkins server. But why do only tests on `test_services_types.py` fail? Are type services heavier than others? Messages like `500 != 204` do not help to understand these errors. Perhaps it is good to print the body of the response when the return code is `500`, to see the backend traceback.

Using my branch `maurits/use-plone-restapi-checkout-52-7xx` I ran these three tests on Python 2:

`bin/test -s plone.restapi -t test_types_document_patch_properties -t test_addable_types_for_non_manager_user -t test_types_document_put`

The middle one does an explicit transaction commit.
Most of the time, all three tests pass, but not always:

```
Failure in test test_types_document_put (plone.restapi.tests.test_services_types.TestServicesTypes)
Traceback (most recent call last):
  File "/Users/maurits/.pyenv/versions/2.7.18/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/Users/maurits/community/plone-coredev/5.2/src/plone.restapi/src/plone/restapi/tests/test_services_types.py", line 407, in test_types_document_put
    self.assertIn("author_name", response.json().get("properties"))
  File "/Users/maurits/.pyenv/versions/2.7.18/lib/python2.7/unittest/case.py", line 804, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/Users/maurits/.pyenv/versions/2.7.18/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: 'author_name' not found in {u'exclude_from_nav': {u'description': u'If selected, this item will not appear in the navigation tree', u'title': u'Exclude from navigation', u'default': False, u'factory': u'Yes/No', u'behavior': u'plone.excludefromnavigation', u'type': u'boolean'}, u'table_of_contents': {u'title': u'Table of contents', u'type': u'boolean', u'description': u'If selected, this will show a table of contents at the top of the page.', u'behavior': u'plone.tableofcontents', u'factory': u'Yes/No'}, u'contributors': {u'additionalItems': True, u'description': u'The names of people that have contributed to this item. Each contributor should be on a separate line.', u'title': u'Contributors', u'items': {u'title': u'', u'type': u'string', u'description': u'', u'factory': u'Text line (String)'}, u'factory': u'Tuple', u'behavior': u'plone.dublincore', u'uniqueItems': True, u'widgetOptions': {u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.Users'}}, u'type': u'array'}, u'effective': {u'widget': u'datetime', u'description': u'If this date is in the future, the content will not show up in listings and searches until this date.', u'title': u'Publishing Date', u'factory': u'Date/Time', u'behavior': u'plone.dublincore', u'type': u'string'}, u'rights': {u'widget': u'textarea', u'description': u'Copyright statement or other rights information on this item.', u'title': u'Rights', u'factory': u'Text', u'behavior': u'plone.dublincore', u'type': u'string'}, u'text': {u'widget': u'richtext', u'description': u'', u'title': u'Text', u'factory': u'Rich Text', u'behavior': u'plone.richtext', u'type': u'string'}, u'expires': {u'widget': u'datetime', u'description': u'When this date is reached, the content will no longer be visible in listings and searches.', u'title': u'Expiration Date', u'factory': u'Date/Time', u'behavior': u'plone.dublincore', u'type': u'string'}, u'allow_discussion': {u'description': u'Allow discussion for this content object.', u'vocabulary': {u'@id': u'http://localhost:54637/plone/@sources/allow_discussion'}, u'title': u'Allow discussion', u'enum': [u'True', u'False'], u'factory': u'Choice', u'choices': [[u'True', u'Yes'], [u'False', u'No']], u'enumNames': [u'Yes', u'No'], u'behavior': u'plone.allowdiscussion', u'type': u'string'}, u'changeNote': {u'title': u'Change Note', u'type': u'string', u'description': u'Enter a comment that describes the changes you made.', u'behavior': u'plone.versioning', u'factory': u'Text line (String)'}, u'language': {u'description': u'', u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.SupportedContentLanguages'}, u'title': u'Language', u'default': u'en', u'factory': u'Choice', u'behavior': u'plone.dublincore', u'type': u'string'}, u'relatedItems': {u'additionalItems': True, u'description': u'', u'title': u'Related Items', u'default': [], u'items': {u'title': u'Related', u'type': u'string', u'description': u'', u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.Catalog'}, u'factory': u'Relation Choice'}, u'factory': u'Relation List', u'behavior': u'plone.relateditems', u'uniqueItems': True, u'widgetOptions': {u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.Catalog'}, u'pattern_options': {u'recentlyUsed': True}}, u'type': u'array'}, u'versioning_enabled': {u'description': u'Enable/disable versioning for this document.', u'title': u'Versioning enabled', u'default': True, u'factory': u'Yes/No', u'behavior': u'plone.versioning', u'type': u'boolean'}, u'title': {u'title': u'Title', u'type': u'string', u'description': u'', u'behavior': u'plone.dublincore', u'factory': u'Text line (String)'}, u'subjects': {u'additionalItems': True, u'description': u'Tags are commonly used for ad-hoc organization of content.', u'title': u'Tags', u'items': {u'title': u'', u'type': u'string', u'description': u'', u'factory': u'Text line (String)'}, u'factory': u'Tuple', u'behavior': u'plone.dublincore', u'uniqueItems': True, u'widgetOptions': {u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.Keywords'}}, u'type': u'array'}, u'creators': {u'additionalItems': True, u'description': u'Persons responsible for creating the content of this item. Please enter a list of user names, one per line. The principal creator should come first.', u'title': u'Creators', u'items': {u'title': u'', u'type': u'string', u'description': u'', u'factory': u'Text line (String)'}, u'factory': u'Tuple', u'behavior': u'plone.dublincore', u'uniqueItems': True, u'widgetOptions': {u'vocabulary': {u'@id': u'http://localhost:54637/plone/@vocabularies/plone.app.vocabularies.Users'}}, u'type': u'array'}, u'id': {u'title': u'Short name', u'type': u'string', u'description': u'This name will be displayed in the URL.', u'behavior': u'plone.shortname', u'factory': u'Text line (String)'}, u'description': {u'widget': u'textarea', u'description': u'Used in item listings and search results.', u'title': u'Summary', u'factory': u'Text', u'behavior': u'plone.dublincore', u'type': u'string'}}
```

--- a/src/plone/restapi/tests/test_services_types.py
+++ b/src/plone/restapi/tests/test_services_types.py
@@ -5,15 +5,32 @@ from plone.app.testing import SITE_OWNER_NAME
 from plone.app.testing import SITE_OWNER_PASSWORD
 from plone.app.testing import TEST_USER_ID
 from plone.restapi.testing import PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+from plone.restapi.testing import PLONE_VERSION
 from plone.restapi.testing import RelativeSession
 
+import six
+import time
 import transaction
 import unittest
+
+
+PYTHON2_PLONE52 = PLONE_VERSION.base_version >= "5.2" and six.PY2
 
 
 class TestServicesTypes(unittest.TestCase):
 
     layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+
+    def sleep_python2_plone52(self, seconds=0.2):
+        """plone.dexterity 2.10+ now uses the _p_mtime variable in the schema names.
+        As this variable is related to date/time and the tests are executed very fast,
+        some schema had the same name, when they shouldn't be. This was causing
+        problems in the tests. So we introduce sleeps in Plone 5.2 with Python 2,
+        so that the schemas don't have the same names. In Python 3 we don't have this
+        problem because the precision of _p_mtime is higher in Python 3.
+        """
+        if PYTHON2_PLONE52:
+            time.sleep(seconds)
 
     def setUp(self):
         self.app = self.layer["app"]
@@ -60,6 +77,7 @@ class TestServicesTypes(unittest.TestCase):
                 "description": "Website of the author",
             },
         )
+        self.sleep_python2_plone52()
 
     def tearDown(self):
         # Remove all custom changed on Document
@@ -114,6 +132,7 @@ class TestServicesTypes(unittest.TestCase):
         )  # noqa
 
     def test_types_document_get_fieldset(self):
+        self.sleep_python2_plone52()
         response = self.api_session.get("/@types/Document/contact_info")
 
         self.assertEqual(response.status_code, 200)
@@ -253,6 +272,7 @@ class TestServicesTypes(unittest.TestCase):
                 "required": False,
             },
         )
+        self.sleep_python2_plone52()
         self.assertEqual(response.status_code, 204)
 
         response = self.api_session.get("/@types/Document/author_email")
@@ -317,6 +337,7 @@ class TestServicesTypes(unittest.TestCase):
                 },
             },
         )
+        self.sleep_python2_plone52(0.4)
         self.assertEqual(response.status_code, 204)
 
         response = self.api_session.get("/@types/Document")
@@ -389,6 +410,7 @@ class TestServicesTypes(unittest.TestCase):
         }
 
         response = self.api_session.put("/@types/Document", json=doc_json)
+        self.sleep_python2_plone52(0.4)
         self.assertEqual(response.status_code, 204)
 
         response = self.api_session.get("/@types/Document")
@@ -429,6 +451,7 @@ class TestServicesTypes(unittest.TestCase):
         response = self.api_session.delete(
             "/@types/Document/author_email",
         )
+        self.sleep_python2_plone52(0.4)
         self.assertEqual(response.status_code, 204)
 
         response = self.api_session.get("/@types/Document")

